File: Person.java
import java.awt.geom.AffineTransform;

public class Person extends Ragdoll {
    public Person() {
        int torsoW = 60, torsoH = 130;
        Sprite torso = new SpriteRectangle(torsoW, torsoH, null);
        torso.transform(AffineTransform.getTranslateInstance(370, 200));
        torso.setCanTranslate(true);

        root = torso;

        /* HEAD */
        int headW = 40, headH = 60;
        Sprite head = new SpriteEllipse(headW, headH, torso);
        head.transform(AffineTransform.getTranslateInstance((torsoW - headW) / 2, 0));
        head.setPivotAround(headW / 2, 0, 50, 0, 180);

        /* LEFT SIDE OF THE BODY*/

        /* UPPER LEFT ARM */
        int ulArmW = 20, ulArmH = 70;
        Sprite ulArm = new SpriteEllipse(ulArmW, ulArmH, torso);
        ulArm.transform(AffineTransform.getTranslateInstance(-ulArmW/2, 0));
        ulArm.setPivotAround(ulArmW / 2, 0, 360, 10);

        /* LOWER LEFT ARM */
        int llArmW = 20, llArmH = 70;
        Sprite llArm = new SpriteEllipse(llArmW, llArmH, ulArm);
        llArm.transform(AffineTransform.getTranslateInstance(0, ulArmH));
        llArm.setPivotAround(llArmW / 2, 0, 135, -5);

        /* LEFT HAND */
        int lHandW = 20, lHandH = 30;
        Sprite lHand = new SpriteEllipse(lHandW, lHandH, llArm);
        lHand.transform(AffineTransform.getTranslateInstance(0, llArmH));
        lHand.setPivotAround(lHandW / 2, 0, 35);

        /* UPPER LEFT LEG */
        int ulLegW = 20, ulLegH = 90;
        Sprite ulLeg = new SpriteEllipse(ulLegW, ulLegH, torso);
        ulLeg.transform(AffineTransform.getTranslateInstance(0, torsoH));
        ulLeg.setPivotAround(ulLegW / 2, 0, 90);
        ulLeg.setCanScale(true);

        /* LOWER LEFT LEG */
        int llLegW = 20, llLegH = 60;
        Sprite llLeg = new SpriteEllipse(llLegW, llLegH, ulLeg);
        llLeg.transform(AffineTransform.getTranslateInstance(0, ulLegH));
        llLeg.setPivotAround(llLegW / 2, 0, 90);
        llLeg.setCanScale(true);
        ulLeg.setSlave(llLeg);

        /* LEFT FOOT */
        int lFootW = 18, lFootH = 40;
        Sprite lFoot = new SpriteEllipse(lFootW, lFootH, llLeg);
        lFoot.transform(AffineTransform.getTranslateInstance(0, llLegH));
        lFoot.setPivotAround(lFootW / 2, 0, 90, 0, 90);

        /* RIGHT SIDE OF THE BODY*/

        /* UPPER RIGHT ARM */
        Sprite urArm = new SpriteEllipse(ulArmW, ulArmH, torso);
        urArm.transform(AffineTransform.getTranslateInstance(torsoW - ulArmW/2, 0));
        urArm.setPivotAround(ulArmW / 2, 0, 360, -10);

        /* LOWER RIGHT ARM */
        Sprite lrArm = new SpriteEllipse(llArmW, llArmH, urArm);
        lrArm.transform(AffineTransform.getTranslateInstance(0, ulArmH));
        lrArm.setPivotAround(llArmW / 2, 0, 135, 5);

        /* RIGHT HAND */
        Sprite rHand = new SpriteEllipse(lHandW, lHandH, lrArm);
        rHand.transform(AffineTransform.getTranslateInstance(0, llArmH));
        rHand.setPivotAround(lHandW / 2, 0, 35);

        /* UPPER RIGHT LEG */
        Sprite urLeg = new SpriteEllipse(ulLegW, ulLegH, torso);
        urLeg.transform(AffineTransform.getTranslateInstance(torsoW - ulLegW, torsoH));
        urLeg.setPivotAround(ulLegW / 2, 0, 90);
        urLeg.setCanScale(true);

        /* LOWER RIGHT LEG */
        Sprite lrLeg = new SpriteEllipse(llLegW, llLegH, urLeg);
        lrLeg.transform(AffineTransform.getTranslateInstance(0, ulLegH));
        lrLeg.setPivotAround(llLegW / 2, 0, 90);
        lrLeg.setCanScale(true);
        urLeg.setSlave(lrLeg);

        /* RIGHT FOOT */
        Sprite rFoot = new SpriteEllipse(lFootW, lFootH, lrLeg);
        rFoot.transform(AffineTransform.getTranslateInstance(0, llLegH));
        rFoot.setPivotAround(lFootW / 2, 0, 90, 0, -90);
    }
}

File: Person.kt
import java.awt.geom.AffineTransform

class Person : Ragdoll() {
    override val root: Sprite

    init {
        val torsoW = 60
        val torsoH = 130
        val torso = SpriteRectangle(torsoW, torsoH, null).apply {
            transform(AffineTransform.getTranslateInstance(370.0, 200.0))
            setCanTranslate(true)
        }

        root = torso

        /* HEAD */
        val headW = 40
        val headH = 60
        val head = SpriteEllipse(headW, headH, torso).apply {
            transform(AffineTransform.getTranslateInstance(((torsoW - headW) / 2).toDouble(), 0.0))
            setPivotAround(headW / 2, 0, 50.0, 0.0, 180.0)
        }

        /* UPPER LEFT LEG */
        val ulLegW = 20
        val ulLegH = 90
        val ulLeg = SpriteEllipse(ulLegW, ulLegH, torso).apply {
            transform(AffineTransform.getTranslateInstance(0.0, torsoH.toDouble()))
            setPivotAround(ulLegW / 2, 0, 90.0)
            setCanScale(true)
        }

        /* LOWER LEFT LEG */
        val llLegW = 20
        val llLegH = 60
        val llLeg = SpriteEllipse(llLegW, llLegH, ulLeg).apply {
            transform(AffineTransform.getTranslateInstance(0.0, ulLegH.toDouble()))
            setPivotAround(llLegW / 2, 0, 90.0)
            setCanScale(true)
        }.also {
            ulLeg.setSlave(it)
        }

        /* LEFT FOOT */
        val lFootW = 18
        val lFootH = 40
        val lFoot = SpriteEllipse(lFootW, lFootH, llLeg).apply {
            transform(AffineTransform.getTranslateInstance(0.0, llLegH.toDouble()))
            setPivotAround(lFootW / 2, 0, 90.0, 0.0, 90.0)
        }

        /* UPPER RIGHT LEG */
        val urLeg = SpriteEllipse(ulLegW, ulLegH, torso).apply {
            transform(AffineTransform.getTranslateInstance((torsoW - ulLegW).toDouble(), torsoH.toDouble()))
            setPivotAround(ulLegW / 2, 0, 90.0)
            setCanScale(true)
        }

        /* LOWER RIGHT LEG */
        val lrLeg = SpriteEllipse(llLegW, llLegH, urLeg).apply {
            transform(AffineTransform.getTranslateInstance(0.0, ulLegH.toDouble()))
            setPivotAround(llLegW / 2, 0, 90.0)
            setCanScale(true)
        }.also {
            urLeg.setSlave(it)
        }

        /* RIGHT FOOT */
        val rFoot = SpriteEllipse(lFootW, lFootH, lrLeg).apply {
            transform(AffineTransform.getTranslateInstance(0.0, llLegH.toDouble()))
            setPivotAround(lFootW / 2, 0, 90.0, 0.0, -90.0)
        }

        /* UPPER LEFT ARM */
        val ulArmW = 20
        val ulArmH = 70
        val ulArm = SpriteEllipse(ulArmW, ulArmH, torso).apply {
            transform(AffineTransform.getTranslateInstance((-ulArmW / 2).toDouble(), 0.0))
            setPivotAround(ulArmW / 2, 0, 360.0, 10.0)
        }

        /* LOWER LEFT ARM */
        val llArmW = 20
        val llArmH = 70
        val llArm = SpriteEllipse(llArmW, llArmH, ulArm).apply {
            transform(AffineTransform.getTranslateInstance(0.0, ulArmH.toDouble()))
            setPivotAround(llArmW / 2, 0, 135.0, -5.0)
        }

        /* LEFT HAND */
        val lHandW = 20
        val lHandH = 30
        val lHand = SpriteEllipse(lHandW, lHandH, llArm).apply {
            transform(AffineTransform.getTranslateInstance(0.0, llArmH.toDouble()))
            setPivotAround(lHandW / 2, 0, 35.0)
        }

        /* UPPER RIGHT ARM */
        val urArm = SpriteEllipse(ulArmW, ulArmH, torso).apply {
            transform(AffineTransform.getTranslateInstance((torsoW - ulArmW / 2).toDouble(), 0.0))
            setPivotAround(ulArmW / 2, 0, 360.0, -10.0)
        }

        /* LOWER RIGHT ARM */
        val lrArm = SpriteEllipse(llArmW, llArmH, urArm).apply {
            transform(AffineTransform.getTranslateInstance(0.0, ulArmH.toDouble()))
            setPivotAround(llArmW / 2, 0, 135.0, 5.0)
        }

        /* RIGHT HAND */
        val rHand = SpriteEllipse(lHandW, lHandH, lrArm).apply {
            transform(AffineTransform.getTranslateInstance(0.0, llArmH.toDouble()))
            setPivotAround(lHandW / 2, 0, 35.0)
        }
    }
}
